feature-governance-impl/src/main/java/io/novafoundation/nova/feature_governance_impl/domain/referendum/tindergov/RealTinderGovInteractor.kt

```diff
@@ -126,19 +129,17 @@ class RealTinderGovInteractor(
         return tinderGovVotingPowerRepository.getVotingPower(metaId, chainId)
     }
 
-    override suspend fun isVotingPowerAvailable(): Boolean {
+    override suspend fun getVotingPowerState(): VotingPowerState {
         val metaAccount = accountRepository.getSelectedMetaAccount()
         val chain = governanceSharedState.chain()
-        return getVotingPower(metaAccount.id, chain.id) != null
-    }
-
-    override suspend fun isSufficientAmountToVote(): Boolean {
-        val metaAccount = accountRepository.getSelectedMetaAccount()
-        val chain = governanceSharedState.chain()
-        val votingPower = getVotingPower(metaAccount.id, chain.id) ?: return false
-        val asset = walletRepository.getAsset(metaAccount.id, chain.utilityAsset) ?: return false
+        val votingPower = getVotingPower(metaAccount.id, chain.id) ?: return VotingPowerState.Empty
+        val asset = assetUseCase.getCurrentAsset()
 
-        return asset.freeInPlanks >= votingPower.amount
+        return if (asset.freeInPlanks >= votingPower.amount) {
```

Comment:
We shouldnt dublicate the fact that "free balance is used for voting". We can extract it to extension like `Asset.availableToVote`. Otherwise it becomes error prone for future changes